src/trackinfoobject.cpp

```diff
@@ -783,79 +781,85 @@ float TrackInfoObject::getCuePoint() {
 }
 
 void TrackInfoObject::slotCueUpdated() {
-    setDirty(true);
+    markDirty();
     emit(cuesUpdated());
 }
 
-Cue* TrackInfoObject::addCue() {
-    //qDebug() << "TrackInfoObject::addCue()";
+CuePointer TrackInfoObject::addCue() {
     QMutexLocker lock(&m_qMutex);
-    Cue* cue = new Cue(m_id);
```

Comment:
Just checked, no delete. So this actually looks like I fixed a leak :)

On 01/10/2016 09:43 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.cpp 
> <https://github.com/mixxxdj/mixxx/pull/852#discussion_r49281127>:
>
> >      QMutexLocker lock(&m_qMutex);
> > -    Cue* cue = new Cue(m_id);
>
> Smart developers use smart pointers ;-)
>
> I just did not find the place where they where deleted in your patch. Is 
> it still there?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/852/files#r49281127>.
>